include/parser/string-parser.hpp

```diff
@@ -620,13 +575,12 @@ static bool parseString(const std::basic_string<CharT>& inputString,
 */
 template <typename CharT, typename OutT>
 static bool parseCharacter(const std::basic_string<CharT>& inputString,
-                           std::vector<OutT>& output,
-                           CompileState& state) {
+                           std::vector<OutT>& output, CompileState& state) {
```

Comment:
I'm not quite sure what the type barrier is supposed to achieve here. Those barriers should be on the type (`StringParserEngine`) itself